src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -263,6 +261,427 @@ static std::vector<size_t> getBlockND(const VectorDims& shape) {
+    squashed_indices_shape[axis] = 1;
+
+    const std::vector<size_t> dataBlockND = getBlockND(data_shape);
+    const std::vector<size_t> indicesBlockND = getBlockND(indices_shape);
+    const size_t dataBlock_axisplus1 = dataBlockND[axis + 1];
+    const size_t indicesBlock_axisplus1 = indicesBlockND[axis + 1];
+
+    // process serially along 'axis' dimension because of data dependency brought by duplicated value in indices
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        splitter(shape_size(squashed_indices_shape), nthr, ithr, start, end);
+        scatter_elements_update::TensorIterator tensorItr(squashed_indices_shape, axis);
+
+        // When *use_init_val* attribute is false, we need to substitute the copied values at target locations with values that
+        // will not affect the particular reduction algorithms.
+        if (!use_init_val) {
+            const auto value = reduction_neutral_value<DataType>(reduction_type);
+            auto offsets = tensorItr.startover(start, dataBlockND, indicesBlockND);
+            for (size_t worker = start; worker < end; worker++) {
+                auto indices_offset = offsets[1];
+                for (size_t idx = 0; idx < index_dim_size; idx++) {
+                    int64_t idxValue =  getIndicesValue(indicesPtr, indices_offset);
+                    if (idxValue < 0) idxValue += data_dim_size;
+                    ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                    dataPtr[offsets[0] + idxValue * dataBlock_axisplus1] = value;
+                    indices_offset += indicesBlock_axisplus1;
+                }
+
+                // increment
+                tensorItr.increment(offsets, dataBlockND, indicesBlockND);
+            }
+        }
+
+        // Apply the Reduce function in an element-wise fashion. For better performance,
+        // when axis is the last dimension, we iterate along axis in the inner loop; otherwise we iterate axis
+        // in the outer loop.
+        auto offsets = tensorItr.startover(start, dataBlockND, indicesBlockND);
+        if (axis == static_cast<int>(updates_rank - 1)) {
+            for (size_t worker = start; worker < end; worker++) {
+                std::unordered_map<size_t, int64_t> mean_reduction_counters;  // (idxValue, num_sums) for current worker
+
+                auto indices_offset = offsets[1];
+                for (size_t idx = 0; idx < index_dim_size; idx++) {
+                    int64_t idxValue =  getIndicesValue(indicesPtr, indices_offset);
+                    if (idxValue < 0) idxValue += data_dim_size;
+                    ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                    auto dst = &dataPtr[offsets[0] + idxValue * dataBlock_axisplus1];
+                    auto src = &updatePtr[indices_offset];
+                    kernel_func(dst, src);
```

Comment:
BTW, have you checked that there is no real funciton call here inside the asm (inplace optimization was successful)?

Reply:
yes, I checked it with @usstq. It  is inplaced without function call.